app/services/submissions_expiration_handler.rb

```diff
@@ -0,0 +1,36 @@
+class SubmissionsExpirationHandler
+  def self.build
+    new(SubmissionRepository.new, SubmissionsInviter.new)
+  end
+
+  def initialize(submission_repository, submissions_inviter)
+    @submission_repository = submission_repository
+    @submissions_inviter = submissions_inviter
+  end
```

Comment:
I find using `self.build` or `self.call` useful when I need to provide some input data and I do not want to play with fake repositories in tests. Example:

```ruby
def self.call(date:, measurements_repository: MeasurementsRepository.new)
  new(input_data: measurements_repository.for_day(date)).call
end

def initialize(input_data:)
  @input_data: input_data
end
```

This way I can easily inject data for my object in tests `Service.new(input_data: [...])`, without creating a fake repo. But, I can also use `Service.call(date: Date.today)` in the other services.

In your case, I don't see value for creating separate `self.build` method. I think you can just write it this way:
```ruby
def initialize(submission_repository: SubmissionRepository.new, submissions_inviter: SubmissionsInviter.new)
  @submission_repository = submission_repository
  @submissions_inviter = submissions_inviter
end
```